services/decoder/protocols/aave-v3/pools/borrow.ts

```diff
@@ -0,0 +1,124 @@
+import { GoldRushDecoder } from "../../../decoder";
+import type {
+    EventDetails,
+    EventTokens,
+    EventType,
+} from "../../../decoder.types";
+import {
+    DECODED_ACTION,
+    DECODED_EVENT_CATEGORY,
+} from "../../../decoder.constants";
+import { decodeEventLog, type Abi } from "viem";
+import ABI from "./../abis/aave-v3.abi.json";
+import { prettifyCurrency } from "@covalenthq/client-sdk";
+import { TimestampParser } from "../../../../../utils/functions";
+
+export const BorrowDecoder = GoldRushDecoder.on(
+    "aave-v3:Borrow",
+    ["eth-mainnet"],
+    ABI as Abi,
+    async (log_event, tx, chain_name, covalent_client): Promise<EventType> => {
+        const { raw_log_data, raw_log_topics } = log_event;
+
+        enum INTEREST_RATE_MODE {
+            "STABLE" = 1,
+            "VARIABLE" = 2,
+        }
+
+        const { args: decoded } = decodeEventLog({
+            abi: ABI,
+            topics: raw_log_topics as [],
+            data: raw_log_data as `0x${string}`,
+            eventName: "Borrow",
+        }) as {
+            eventName: "Borrow";
+            args: {
+                reserve: string;
+                user: string;
+                onBehalfOf: string;
+                amount: bigint;
+                interestRateMode: INTEREST_RATE_MODE;
+                borrowRate: bigint;
+                referalCode: bigint;
+            };
+        };
+
+        const details: EventDetails = [
+            {
+                title: "Reserve",
+                value: decoded.reserve,
+                type: "address",
+            },
+            {
+                title: "User",
+                value: decoded.user,
+                type: "address",
+            },
+            {
+                title: "On Behalf Of",
+                value: decoded.onBehalfOf,
+                type: "address",
+            },
+            {
+                title: "Borrow Rate",
+                value: String(decoded.borrowRate),
+                type: "text",
```

Comment:
The Borrow Rate seems like gibberish - try to calculate what its actual value stands for and parse it accordingly